app/src/main/java/org/mozilla/fenix/home/intent/CrashReporterIntentProcessor.kt

```diff
@@ -5,27 +5,32 @@
 package org.mozilla.fenix.home.intent
 
 import android.content.Intent
+import android.util.Log
 import androidx.navigation.NavController
 import mozilla.components.lib.crash.Crash
-import org.mozilla.fenix.NavGraphDirections
+import mozilla.components.lib.crash.Crash.NativeCodeCrash
+import org.mozilla.fenix.components.AppStore
+import org.mozilla.fenix.components.appstate.AppAction
+import mozilla.components.lib.crash.CrashReporter
 
 /**
- * When the app crashes, the user has the option to report it.
- * Reporting fires an intent to the main activity which is handled here.
+ * Process the [Intent] from [CrashReporter] through which the app is informed about
+ * recoverable native crashes.
  */
-class CrashReporterIntentProcessor : HomeIntentProcessor {
+class CrashReporterIntentProcessor(private val store: AppStore) : HomeIntentProcessor {
 
     override fun process(intent: Intent, navController: NavController, out: Intent): Boolean {
         return if (Crash.isCrashIntent(intent)) {
-            openToCrashReporter(intent, navController)
+            val crash = Crash.fromIntent(intent)
+            if ((crash as? NativeCodeCrash)?.isFatal == false) {
+                store.dispatch(AppAction.AddNonFatalCrash(crash))
+            } else {
+                // Note that an unexpected crash was sent but avoid leaking potential sensitive information.
```

Comment:
We had some work here last week remembering/figuring out the details. Would you mind adding some more comments/info here why both fatal and non native crashed don't end up here. Thanks!

Reply:
Added
```
A fatal crash means the app's main process is affected.
An UncaughtExceptionCrash refers to a [Throwable] that would otherwise crash the app
but is intercepted to allow us to gather more info and crash more gracefully.

In both cases the app is left in a bad state so the main process is killed
but not before gathering more info about the crashes to form and persist a crash report and
not before "CrashHandlerService" is started in a separate process to be able to
show a dialog allowing users to send the crash report and maybe restart the app.
```

Reply:
@csadilek Tried to offer a bit of context about those cases, how they get the app in such a bad state that the main process has to be killed with us then relying on `CrashHandlerService`.
Would adding
> So in these cases where the app cannot be used anymore we don't expect an
Intent for an in-app crash reporter.

help?

Reply:
Yes, that's good! Thanks. I just wanted us to point out how fatal crashes are handled and why they don't end up here. So goo now!